tests/eas/load_tracking.py

```diff
@@ -1293,3 +1294,119 @@ def test_util_tg4_group(self):
 
+
+class CgroupsMigrationTest(_PELTTaskGroupsTest):
+    """
+    Test PELT utilization for task groups migration.
+    Initial group hierarchy:
+                           +-----+
+                           | "/" |
+                           +-----+
+                           /     \
+                          /       \
+                      +------+   +------+
+                      |"/tg1"|   |"/tg2"|
+                      +------+   +------+
+                       /    \         \
+                      /      \         \
+                    t1_1    t1_2      t2_1
+
+    Final group hierarchy:
+                           +-----+
+                           | "/" |
+                           +-----+
+                           /     \
+                          /       \
+                      +------+   +------+
+                      |"/tg1"|   |"/tg2"|
+                      +------+   +------+
+                       /          /    \
+                      /          /      \
+                    t1_1       t1_2    t2_1
+    """
+    target_cpu = 0
+    root_group = None
+    trace = None
+
+    @classmethod
+    def _getExperimentsConf(cls, test_env):
+        # Run all workloads on a CPU with highest capacity
+        cls.target_cpu = min(test_env.calibration(),
+                             key=test_env.calibration().get)
+
+        # Create taskgroups
+        cpus = test_env.target.list_online_cpus()
+        mems = 0
+        cls.root_group = Taskgroup("/", cpus, mems, test_env)
+        tg1 = Taskgroup("/tg1", cpus, mems, test_env)
+        tg2 = Taskgroup("/tg2", cpus, mems, test_env)
+
+        # Create tasks
+        period_ms = 16
+        duration_s = 4
+        cpus = [cls.target_cpu]
+        t1_1 = Task("t1_1", test_env, cpus, period_ms=period_ms,
+                    duty_cycle_pct=10, duration_s=duration_s)
+        t1_2 = Task("t1_2", test_env, cpus, period_ms=period_ms,
+                    duty_cycle_pct=20, duration_s=duration_s)
+        t2_1 = Task("t2_1", test_env, cpus, period_ms=period_ms,
+                    duty_cycle_pct=5, duration_s=duration_s)
+
+        # Link nodes to the hierarchy tree
+        cls.root_group.add_children([tg1, tg2])
+        tg1.add_children([t1_1, t1_2])
+        tg2.add_children([t2_1])
+
+        conf = {
+            'tag' : 'cgp_migration',
+            'flags' : ['ftrace', 'freeze_userspace'],
+            'cpufreq' : {'governor' : 'performance'},
+        }
+
+        return {
+            'wloads': {},
+            'confs': [conf],
+        }
+
+    @classmethod
+    def setUpClass(cls, *args, **kwargs):
+        super(CgroupsMigrationTest, cls).runExperiments(*args, **kwargs)
+
+    @classmethod
+    def _migrate_task(cls, test_env):
+        task = None
+        new_taskgroup = None
+        exclude = ''
+        cgroups = test_env.target.cgroups
+
+        # Get the task and taskgroup for the migration
+        for se in cls.root_group.iter_nodes():
```

Comment:
This seems more like a `class Task(SchedEntity)` internals... cannot we fold these bits into the implementation of the `change_taskgroup` method, in such a way that we have a more easy API for test writers where you just specify that you wanna move a given task `t1_2` into a given group `/tg2`?

Reply:
Done: I added a get_children function in Task(SchedEntity) which take a name in argument and return the task object associate to this name such that the loop to find the object is coded into the Task class